src/main/java/se/liu/ida/hefquin/engine/wrappers/graphqlwrapper/data/GraphQLEntrypoint.java

```diff
@@ -0,0 +1,21 @@
+package se.liu.ida.hefquin.engine.wrappers.graphqlwrapper.data;
+
+import java.util.TreeMap;
+
+public interface GraphQLEntrypoint {
+    /**
+     * @return the GraphQL query fieldname
+     */
+    public String getFieldName();
+
+    /**
+     * @return the parameter definitions for the entrypoint as a map
+     * consisting of parameternames mapped to parameter value types.
+     */
+    public TreeMap<String,String> getParameterDefinitions();
```

Comment:
Why do you need the concrete type `TreeMap`? Please consider using the interface `Map` as the return type.

Reply:
The reason for why I changed to TreeMap is because without the guaranteed ordering of the keys in the keyset of the TreeMap, the GraphQLQueryImpl .toString() method might return different answers each time. The toString method in the GraphQLQueryImpl adds the parameters of the query through iterating over the keyset. By having a normal Map or HashMap the ordering of how these parameters are added can be different. This makes it difficult to create a testcase since when comparing the translated query to the expected "hard coded" query, the translated query might be different from time to time.

Perhaps I should change to the `SortedMap` interface?

Reply:
I have two follow-up comments/questions:
1. Is it completely necessary to have the `toString` method to be deterministic? I'm worried that we are adding computing time to keep sorted something that isn't 100% required
2. If it is necessary to keep the map sorted, yes, please use the interface `SortedMap` as a return type. Otherwise, just use `Map`.